roles/firewall/commit/tasks/main.yml

```diff
@@ -1,20 +1,24 @@
 ---
 - name: Enable service ports via iptables
+  delegate_to: "{{ item.delegate|default(omit) }}"
```

Comment:
I think this is the wrong place to perform the delegation.  This should happen when roles modify the `firewall_data` value.  This may require both delegate_to and delegate_facts.